### PR TITLE
[misc] soft drop beam search

### PR DESCRIPTION
**Pull request by youkaichao**
manually tested, this will not result in DoS. a single beam search request will get "bad request", while the rest requests are processed as normal.

server side:

```bash
$ vllm serve meta-llama/Meta-Llama-3-8B
```

client side:

```python
from openai import OpenAI

client = OpenAI(
    base_url="http://localhost:8000/v1",
    api_key="key123",
)

prompt = "Capital of France is"

try:
    completion = client.completions.create(
        model="meta-llama/Meta-Llama-3-8B",
        prompt=prompt,
        max_tokens=4,
        extra_body={'use_beam_search': True, 'best_of': 3}
    )
except Exception as e:
    print(e)

completion = client.completions.create(
    model="meta-llama/Meta-Llama-3-8B",
    prompt=prompt,
    max_tokens=4,
)

print(completion.choices[0].text)
```

output:

```text
Error code: 400 - {'object': 'error', 'message': 'Using beam search as a sampling parameter is deprecated, and will be removed in the future release. Please use the `vllm.LLM.use_beam_search` method for dedicated beam search instead, or set the environment variable `VLLM_ALLOW_DEPRECATED_BEAM_SEARCH=1` to suppress this error. For more details, see https://github.com/vllm-project/vllm/issues/8306 .', 'type': 'BadRequestError', 'param': None, 'code': 400}
 Paris and its official
```

### CODE REVIEW

**Comment by youkaichao** [comment]
ci should be fine because I just merged https://github.com/vllm-project/buildkite-ci/pull/34

File: vllm/sampling_params.py [hunk 1]
```diff
@@ -260,6 +261,10 @@ def __post_init__(self) -> None:
 
         self._verify_args()
         if self.use_beam_search:
+            if not envs.VLLM_ALLOW_DEPRECATED_BEAM_SEARCH:
+                raise ValueError(
+                    "Using beam search as a sampling parameter is deprecated, and will be removed in the future release. Please use the `vllm.LLM.use_beam_search` method for dedicated beam search instead, or set the environment variable `VLLM_ALLOW_DEPRECATED_BEAM_SEARCH=1` to suppress this error. For more details, see https://github.com/vllm-project/vllm/issues/8306 ."  # noqa
```
Comment (KuntaiDu) [inline vllm/sampling_params.py:266]: Do we provide openAI-compatible server alternative for beam search (not sure if this is a valid need and feel free to ignore this comment)?

Comment (youkaichao) [reply vllm/sampling_params.py:266]: no. see the future work section at https://github.com/vllm-project/vllm/pull/8726#issuecomment-2368872745

**Review by comaniac — APPROVED** [review]
LGTM. One question I have is this still allows users to use the beam search we are going to deprecate, meaning that we cannot remove beam search in the next PR. So are we planning to remove it after this release?

**Comment by youkaichao** [comment]
merge as failed tests are irrelevant.

**Pull request merged**